qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -280,6 +280,21 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         return self._run(dag, tracker)
 
     def _run(self, dag: DAGCircuit, tracker: QubitTracker) -> DAGCircuit:
+        # Check if HighLevelSynthesis can be skipped.
+        all_skipped = True
+        for node in dag.topological_op_nodes():
```

Comment:
We technically don't need to waste time getting a topological order for this function, we can just look at every op node in arbitrary order.